### PR TITLE
Use precompiled struct methods for pack, unpack data

### PR DESCRIPTION
**Pull request by umax**
Notes for struct usage optimizations: https://docs.python.org/2/library/struct.html#classes
Regarding string formatting:

```
In [1]: %timeit '%s' % 1
14.7 ns ± 0.325 ns per loop (mean ± std. dev. of 7 runs, 100000000 loops each)

In [2]: %timeit '{0}'.format(1)
134 ns ± 2.93 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

### CODE REVIEW

File: prometheus_client/core.py [hunk 1]
```diff
@@ -29,6 +29,11 @@
 _MINUS_INF = float("-inf")
 _INITIAL_MMAP_SIZE = 1 << 20
 
+pack_integer = struct.Struct(b'i').pack_into
```
Comment (brian-brazil) [inline prometheus_client/core.py]: put an _ in front of these so they aren't exported

File: prometheus_client/core.py [hunk 2]
```diff
@@ -376,19 +381,26 @@ def _init_value(self, key):
...
         self._positions[key] = self._used - 8
 
     def _read_all_values(self):
         """Yield (key, value, pos). No locking is performed."""
+
         pos = 8
-        while pos < self._used:
-            encoded_len = struct.unpack_from(b'i', self._m, pos)[0]
+
+        # performance tricks
```
Comment (brian-brazil) [inline prometheus_client/core.py]: Explain what you're doing a bit more. Not everyone will have seen the relevant talks by Raymond Hettinger

Comment (umax) [reply prometheus_client/core.py]: https://wiki.python.org/moin/PythonSpeed/PerformanceTips

File: prometheus_client/core.py (hunk 1 above)
Comment (umax) [reply prometheus_client/core.py]: ok

File: prometheus_client/core.py (hunk 2 above)
Comment (brian-brazil) [reply prometheus_client/core.py]: I know why this works, but someone reading this code might not. A quick comment will help future readers of the code.

Comment (umax) [reply prometheus_client/core.py]: fixed

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**